TradeItIosTicketSDK2/TradeItLinkedBrokerCache.swift

```diff
@@ -67,112 +56,58 @@ class TradeItLinkedBrokerCache {
-                fxBalance.buyingPowerBaseCurrency = fxBuyingPower
-
-                account.fxBalance = fxBalance
-            }
-
-            return account
+extension CachedLinkedBroker {
+    
+    convenience init(linkedBroker: TradeItLinkedBroker) {
+        self.init()
+        self.accounts = linkedBroker.accounts.map { account in
+            return CachedLinkedBrokerAccount(linkedBrokerAccount: account)
         }
+        self.accountsLastUpdated = linkedBroker.accountsLastUpdated
+        self.isAccountLinkDelayedError = linkedBroker.isAccountLinkDelayedError
     }
+}
 
-    private func serialize(accounts: [TradeItLinkedBrokerAccount]) -> [SerializedLinkedBrokerAccount] {
-        var serializeAccountsList: [SerializedLinkedBrokerAccount] = []
-        for account in accounts {
-            var serializedAccount = SerializedLinkedBrokerAccount()
-            serializedAccount[ACCOUNT_NAME_KEY] = account.accountName
-            serializedAccount[ACCOUNT_NUMBER_KEY] = account.accountNumber
-            serializedAccount[ACCOUNT_INDEX_KEY] = account.accountIndex
-            serializedAccount[ACCOUNT_BASE_CURRENCY_KEY] = account.accountBaseCurrency
-            serializedAccount[ACCOUNT_ENABLED_KEY] = account.isEnabled ? ACCOUNT_ENABLED_VALUE : ACCOUNT_DISABLED_VALUE
-
-            if let balance = account.balance,
-                let buyingPower = balance.buyingPower {
-
-                if let balanceLastUpdated = account.balanceLastUpdated {
-                    serializedAccount[BALANCE_LAST_UPDATED_KEY] = balanceLastUpdated
-                }
-
-                serializedAccount[BALANCE_BUYING_POWER_KEY] = buyingPower
-            }
-
-            if let fxBalance = account.fxBalance,
-                let buyingPowerBaseCurrency = fxBalance.buyingPowerBaseCurrency {
-
-                if let balanceLastUpdated = account.balanceLastUpdated {
-                    serializedAccount[BALANCE_LAST_UPDATED_KEY] = balanceLastUpdated
-                }
+extension CachedLinkedBrokerAccount {
+    convenience init(linkedBrokerAccount: TradeItLinkedBrokerAccount) {
+        self.init()
+        self.accountNumber = linkedBrokerAccount.accountNumber
+        self.accountName = linkedBrokerAccount.accountName
+        self.accountBaseCurrency = linkedBrokerAccount.accountBaseCurrency
+        self.accountIndex = linkedBrokerAccount.accountIndex
+        self.isEnabled = linkedBrokerAccount.isEnabled
+        self.balance = CachedAccountOverview(balance: linkedBrokerAccount.balance)
+        self.fxBalance = CachedFxAccountOverview(fxBalance: linkedBrokerAccount.fxBalance)
+        self.balanceLastUpdated = linkedBrokerAccount.balanceLastUpdated
+    }
+}
 
-                serializedAccount[FX_BALANCE_BUYING_POWER_KEY] = buyingPowerBaseCurrency
-            }
+extension CachedAccountOverview {
+    convenience init(balance: TradeItAccountOverview?) {
+        self.init()
+        self.buyingPower = balance?.buyingPower
```

Comment:
Since we are serializing to JSON and don't need to create new keys in NSUserDefaults anymore to store new data, should we just cache all the balance fields instead of just `buyingPower`?